crates/kubelet/src/server.rs

```diff
@@ -122,11 +122,14 @@ async fn get_container_logs<T: Provider + Sync>(
     }
     match provider.logs(namespace, pod, container).await {
         Ok(data) => Response::new(Body::from(data)),
-        // TODO: This should detect not implemented vs. regular error
         Err(e) => {
             error!("Error fetching logs: {}", e);
-            let mut res = Response::new(Body::from("Not Implemented"));
-            *res.status_mut() = StatusCode::NOT_IMPLEMENTED;
+            let mut res = Response::new(Body::from(format!("Server error: {}", e)));
```

Comment:
Nit: This and the status should probably use the builder patter like we are doing elsewhere

Reply:
Do you have a link to an example? Still wading my way around the codebase